source/server/listener_manager_impl.cc

```diff
@@ -108,33 +109,67 @@ std::vector<Network::FilterFactoryCb> ProdListenerComponentFactory::createNetwor
   return ret;
 }
 
-std::vector<Network::ListenerFilterFactoryCb>
-ProdListenerComponentFactory::createListenerFilterFactoryList_(
+Filter::ListenerFilterFactoriesList
+ProdListenerComponentFactory::createListenerFilterFactoryListImpl(
     const Protobuf::RepeatedPtrField<envoy::config::listener::v3::ListenerFilter>& filters,
-    Configuration::ListenerFactoryContext& context) {
-  std::vector<Network::ListenerFilterFactoryCb> ret;
+    Configuration::ListenerFactoryContext& context,
+    Filter::TcpListenerFilterConfigProviderManagerImpl& config_provider_manager) {
+  Filter::ListenerFilterFactoriesList ret;
+
+  ret.reserve(filters.size());
   for (ssize_t i = 0; i < filters.size(); i++) {
     const auto& proto_config = filters[i];
     ENVOY_LOG(debug, "  filter #{}:", i);
     ENVOY_LOG(debug, "    name: {}", proto_config.name());
     ENVOY_LOG(debug, "  config: {}",
               MessageUtil::getJsonStringFromMessageOrError(
                   static_cast<const Protobuf::Message&>(proto_config.typed_config())));
-
-    // Now see if there is a factory that will accept the config.
-    auto& factory =
-        Config::Utility::getAndCheckFactory<Configuration::NamedListenerFilterConfigFactory>(
-            proto_config);
-    auto message = Config::Utility::translateToFactoryConfig(
-        proto_config, context.messageValidationVisitor(), factory);
-    ret.push_back(factory.createListenerFilterFactoryFromProto(
-        *message, createListenerFilterMatcher(proto_config), context));
+    // dynamic listener filter configuration
+    if (proto_config.config_type_case() ==
```

Comment:
Can this piece of code be generalized with the [HCM `procesFilter`](https://github.com/envoyproxy/envoy/blob/6d85d09c7dbc7c3a86dddbe89a35c54f7c431cac/source/extensions/filters/network/http_connection_manager/config.cc#L622) code?

Reply:
The code has similarities. But there are some subtle differences. I would rather not mingle them together.

Reply:
I guess this also relates to this [previous comment](https://github.com/envoyproxy/envoy/pull/21133/files#r868019075).
Seems that the design could be improved here.

Reply:
Leaving this to the item 8 of issue #20049.

Reply:
will address this in the coming UDP listener ECDS work as in the item 8 of issue: https://github.com/envoyproxy/envoy/issues/20049.